gprofiler/profilers/registry.py

```diff
@@ -0,0 +1,76 @@
+from typing import Any, Dict, List, Optional, Type
+
+
+class ProfilerArgument:
+    def __init__(
+        self,
+        name: str,
+        dest: str,
+        help: str,
+        default: Any = None,
+        action: Optional[str] = None,
+        choices: List[Any] = None,
+        type: Type = None,
+    ):
+        self.name = name
+        self.dest = dest
+        self.help = help
+        self.default = default
+        self.action = action
+        self.choices = choices
+        self.type = type
+
+    def get_dict(self) -> Dict[str, Any]:
+        dict_argument_names = ["help", "default", "action", "choices", "dest", "type"]
+        argument_dict = {}
+        for argument_name in dict_argument_names:
+            argument_value = getattr(self, argument_name)
+            if argument_value is not None:
+                argument_dict[argument_name] = argument_value
+        return argument_dict
+
+
+class ProfilerConfig:
+    def __init__(
+        self,
+        profiler_mode_help: str,
+        possible_modes: List[str] = None,
+        default_mode: str = "auto",
```

Comment:
Any reason for the default (of this & the others?)

Reply:
Fixed and changed to "enabled"